CHANGELOG.md

```diff
@@ -1,236 +1,265 @@
-A-device-plugin will put its usable device and its status in "Node-A-device-register" annotation
-scheduler will read from this annotation and acknowledge this registration. So, gRpc will no longer be used.
-
-**Optimization in code**
-
-Put nvidia-device-plugin related code in a seperate directory "nvidiadevice"
-
-**Libvgpu log adjusting**
-
-Downgrade the following API from LOG:WARN to LOG:INFO
-cuFuncSetCacheConfig, cuFuncSetCacheConfig ,cuModuleGetTexRef, cuModuleGetSurfRef
-
-## v2.2.10
-
-**BUG fix**
-
-fix process can't initialize properly in driver 440
-
-fix cuCtxCreate failed in some tensorRT task
-
-fix env CUDA_VISIBLE_DEVICES not working properly sometimes.
-
-## v2.2.11
-
-**BUG fix**
-
-fix initialization failed with non-root uers
-
-fix core limitation not working properly on A30
-
-## v2.2.12
-
-Downgrade core control output from LOG:WARN to LOG:DEBUG
-
-## v2.2.13
-
-Adjust default memory to 0, which means use 100% device memory
-
-Move cache file directory from /tmp/vgpu/containers to /usr/local/vgpu/containers
-
-## v2.2.14
-
-Fix device memory calculation error after container crashloop
-
-Fix env cuda_oversubscribe not set properly when MemoryScaling < 1
-
-Fix MemoryScaling not working when set < 1
-
-## v2.2.15
-
-Move shared-memory from from /tmp/xxx.cache to /usr/local/vgpu/xxx.cache inside container
-
-Add Deviceidx to scheduler monitor apis(31993)
-
-## v2.2.16
-
-Fix crash during initlization in vGPUmonitor
-
-# v2.3
-
-## v2.3.0
-
-Fix oom can't be triggered when loading module
-
-Update device-plugin version to v0.14.0
-
-## v2.3.1
-
-Fix a bug where a cuda process can't be launched properly
-
-## v2.3.2
-
-Remove node selector for scheduler
-
-Fix an issue where mlu device-plugin can't be launched properly
-
-Major rework on devices-related code
-
-Add support for hygon DCU device
-
-## v2.3.3
-
-Fix an issue where pod pending on nodes with multi-architect devices.
-
-## v2.3.4
-
-Fix an issue where 31993 port can't list all GPU nodes
-
-Add a switch on cuda_control by set env "CUDA_DISABLE_ENV=true" in container
-
-## v2.3.6
-
-Fix initialization error when using ray
-
-## v2.3.7
-
-Fix error when "N/A" is shown in command "nvidia-smi topo -m"
-Fix core utilization not working on some cases
-Adjust some documents
-
-## v2.3.8
-
-Fix device-plugin launch error on driver version < 500
-
-support manual config MutatingWebhookConfiguration failurePolicy
-
-add metrics bind address flag for scheduler
-
-Improved log messages
-
-fix: loss of metrics after vdeivce restart
-
-bugfix: device-plugin monitor serves too slowly in big cluster
-
-## v2.3.9
-
-Add support for iluvatar GPU devices
-
-Fix issue on "get_host_pid" func in HAMi-core
-
-Regular devices API, make it easier to add new devices
-
-## v2.3.10
-
-Fix issue where device-plugin failed to start
-
-## v2.3.11
-
-Add support for Ascend910B device
-
-Add "NVIDIA_VISIBLE_DEVICES=none" to none-gpu tasks
-
-
+# CHANGELOG
+
+## v2.4.1 - 2024/11/05

```

Comment:
The current changelog is automatically generated by github ci pipeline

We can try to optimize ci and dynamically persist it to a file

Reply:
agree, then to do it, before this change log file caused more confusion.